transferFiles-http/index.js

```diff
@@ -9,28 +9,75 @@ const { Storage } = require('@google-cloud/storage');
 exports.transferFiles = async (req, res) => {
   const storage = new Storage();
 
-  const srcBucket = req.body.srcBucket || 'gcp-bucket-deposit';
-  const destBucket = req.body.destBucket || 'gcp-bucket-destination';
+  const srcBucketName = req.body.srcBucket || 'gcp-bucket-deposit';
+  const destBucketName = req.body.destBucket || 'gcp-bucket-destination';
   const deleteSrc = req.body.deleteSrc || false;
 
-  const [files] = await storage.bucket(srcBucket).getFiles();
+  // check if buckets exist
+  let srcBucket, srcBucketExists, destBucket, destBucketExists;
+  try {
+    srcBucketExists = await storage.bucket(srcBucketName).exists();
 
-  console.log(`files: ${files}`);
+    if (srcBucketExists[0]) {
+      srcBucket = storage.bucket(srcBucketName);
+    } else {
+      res.status(404).send(`srcBucket: ${srcBucket} not found`);
+    }
+  } catch (err) {
+    console.error(err.message);
+    res.status(401).send(`Unauthorized to access ${srcBucketName}`);
+  }
+
+  try {
+    destBucketExists = await storage.bucket(destBucketName).exists();
+
+    if (destBucketExists[0]) {
+      destBucket = storage.bucket(destBucketName);
+    } else {
+      res.status(404).send(`destBucket: ${destBucket} not found`);
+    }
+  } catch (err) {
+    console.error(err.message);
+    res.status(401).send(`Unauthorized to access ${srcBucketName}`);
+  }
+
+  // iterate through all files in srcBucket
+  const [srcFiles] = await srcBucket.getFiles();
 
-  files.forEach(async (file) => {
+  for await (file of srcFiles) {
```

Comment:
Is an await needed here? srcFiles should be an array of files already. the for await could still be used like so:`for await (file of srcBucket.getFiles())` or the callback on the getFiles function could be used: `await srcBucket.getFiles({}, (err, files) => { 
// do stuff here
})`

Reply:
fixed